StepsToCare/CareDashboard.ascx.cs

```diff
@@ -2683,31 +2898,49 @@ private void CheckAndAddQuickNote()
                             mdConfirmNote.Show();
                         }
                     }
+                    else if ( quickNoteId.Value == -1 && careNeed != null )
+                    {
+                        gMakeNote_Click( null, new RowEventArgs( 0, careNeed.Id ) );
+                    }
                 }
 
             }
         }
 
-        private void SnoozeNeed( int id )
+        private void SnoozeNeed( int id, DateTime? selectedDateTime = null )
         {
             using ( var rockContext = new RockContext() )
             {
                 var snoozeChildNeeds = GetAttributeValue( AttributeKey.SnoozeChildNeeds ).AsBoolean();
                 var careNeedService = new CareNeedService( rockContext );
                 var careNeed = careNeedService.Get( id );
                 var snoozeValueId = DefinedValueCache.Get( rocks.kfs.StepsToCare.SystemGuid.DefinedValue.CARE_NEED_STATUS_SNOOZED ).Id;
-                if ( careNeed.StatusValueId != snoozeValueId && ( !careNeed.RenewMaxCount.HasValue || careNeed.RenewCurrentCount <= careNeed.RenewMaxCount.Value ) )
+                if ( careNeed.StatusValueId != snoozeValueId && ( !careNeed.RenewMaxCount.HasValue || careNeed.RenewCurrentCount <= careNeed.RenewMaxCount.Value || selectedDateTime != null ) )
```

Comment:
Is `!careNeed.RenewMaxCount.HasValue || careNeed.RenewCurrentCount <= careNeed.RenewMaxCount.Value` still needed?